yt_dlp/postprocessor/embedthumbnail.py

```diff
@@ -119,14 +119,21 @@ def run(self, info):
             if not mutagen or prefer_atomicparsley:
                 success = False
             else:
+                self._report_run('mutagen', filename)
+                f = {'jpeg': MP4Cover.FORMAT_JPEG, 'png': MP4Cover.FORMAT_PNG}
                 try:
-                    self._report_run('mutagen', filename)
+                    with open(thumbnail_filename, 'rb') as thumbfile:
+                        thumb_data = thumbfile.read()
+
+                    type_ = imghdr.what(h=thumb_data)
+                    if not type_:
+                        raise ValueError('could not determine image type')
+                    elif type_ not in f:
+                        raise ValueError(f'incompatible image type: {type_}')
```

Comment:
should this be done earlier, e.g. around L84 where we convert to png if the thumb isn't a compatible format?

Reply:
I think given that `imghdr` works as expected that branch should not be hit, but you never know. The discrepancy only happens if you use `imghdr`, so i think doing it in this branch is fine